portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,7 +35,46 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Joachim!</h1>");
+    Gson gson = new Gson();
+
+    ArrayList<String> data = new ArrayList<String>(0);
+    Query query = new Query("Comments");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+      data.add(comment);
+    }
+
+    String json = gson.toJson(data);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = getComment(request);
+
+    Entity commentEntity = new Entity("Comments");
+    commentEntity.setProperty("comment", comment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // data.add(0,comment);
+
+    datastore.put(commentEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  private String getComment(HttpServletRequest request) {
+    String full_String = request.getParameter("name");
```

Comment:
It's best to stay consistent with your naming scheme.  The rest of the variables and methods are using camelCase, so stick with that instead of introducing snake_case.